status/status.go

```diff
@@ -244,3 +294,153 @@ func printJSON(all []se) {
+	Self         string
+}
+
+type server struct {
+	response serveResponse
+	mu       sync.Mutex
+
+	si       *network.ServerIdentity
+	list     []*network.ServerIdentity
+	timeout  time.Duration
+	interval time.Duration
+	quit     chan struct{}
+}
+
+func serve(c *cli.Context) error {
+	if c.NArg() != 2 {
+		return errors.New("please give 2 arguments: group.toml private.toml")
+	}
+	ro, err := readGroup(c.Args().First())
+	if err != nil {
+		return errors.New("couldn't read file: " + err.Error())
+	}
+	log.Lvl3(ro)
+	list := ro.List
+	log.Info("List is", list)
+	to, err := time.ParseDuration(c.Parent().String("timeout"))
+	if err != nil {
+		return errors.New("duration parse error: " + err.Error())
+	}
+	coth, err := app.LoadCothority(c.Args().Get(1))
+	if err != nil {
+		return errors.New("error while loading private.toml: " + err.Error())
+	}
+	si, err := coth.GetServerIdentity()
+	if err != nil {
+		return errors.New("private.toml didn't have a serverIdentity: " + err.Error())
+	}
+
+	format := c.String("format")
+	if format != "prometheus" {
+		return errors.New("unsupported format: " + format)
+	}
+
+	endpoint := c.String("endpoint")
+	if len(endpoint) == 0 || endpoint[0] != '/' {
+		return errors.New("invalid endpoint: " + endpoint)
+	}
+
+	port := c.Int("port")
+	interval := c.Int("interval")
+
+	s := &server{
+		interval: time.Duration(interval) * time.Second,
+		timeout:  to,
+		list:     list,
+		si:       si,
+		quit:     make(chan struct{}),
+	}
+
+	http.HandleFunc(endpoint, s.serveHandler)
+	httpServer := &http.Server{Addr: fmt.Sprintf(":%d", port)}
+
+	done := s.probe()
+	go func() {
+		log.Infof("Starting HTTP server on %d\n", port)
+		if err := httpServer.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			log.Errorf("error starting server")
+			return
+		}
+	}()
+
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
+
+	<-signalChan
+	log.Infof("Shutting down server..")
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	if err := httpServer.Shutdown(ctx); err != nil {
+		log.Errorf("Error shutting down server: %s", err)
+	}
+
+	s.quit <- struct{}{}
+	<-done
+	return nil
+}
+
+func (s *server) probe() chan struct{} {
+	done := make(chan struct{})
+	go func() {
+		tick := time.Tick(s.interval)
+		for {
+			select {
+			case <-s.quit:
+				log.Infof("Shutting down prober...")
+				done <- struct{}{}
+				return
+			case <-tick:
+				log.Infof("Probing servers...")
+				s.mu.Lock()
+				s.response = serveResponse{Matrix: make(map[string]int)}
+				s.response.Connectivity = 0
+				s.response.Self = s.si.Address.String()
+				for _, si := range s.list {
+					s.response.Matrix[si.String()] = 0
+				}
+
+				resp, err := status.NewClient().CheckConnectivity(
+					s.si.GetPrivate(),
+					s.list,
+					s.timeout,
+					true,
+				)
+
+				if err != nil {
+					log.Errorf("error checking connectivity: %s", err)
```

Comment:
For debugging purposes, I'm wondering:
should we output this as a comment on the prometheus page?
It might help DEDIS debug problems without having access to the VPSI server.